sled-agent/src/storage_manager.rs

```diff
@@ -40,6 +43,10 @@ use illumos_utils::{zfs::MockZfs as Zfs, zpool::MockZpool as Zpool};
 #[cfg(not(test))]
 use illumos_utils::{zfs::Zfs, zpool::Zpool};
 
+// A key manager can only become ready once. This occurs during RSS or cold
+// boot when the bootstore has detected it has a key share.
+static KEY_MANAGER_READY: OnceLock<bool> = OnceLock::new();
```

Comment:
Nit - is it possible for this to be set to `false`? If not and we're only using this as a threadsafe "set or not", could we do use `OnceLock<()>` instead, and the set-ness of the lock itself is whether it's ready?

Reply:
Oh, I was thinking about this, but I always forget the unit type exists! Thanks for pointing this out.  It's a definite change to make.

Reply:
Fixed in [f4d62ad](https://github.com/oxidecomputer/omicron/pull/3800/commits/f4d62ad6e08e39f6fd8f9f192ce8b19916408349)